src/shared-types/src/AuditLogDynamoGateway.ts

```diff
@@ -1,12 +1,19 @@
-import type { AuditLog, AuditLogEvent, PromiseResult } from "."
+import type { DocumentClient } from "aws-sdk/clients/dynamodb"
+import type { AuditLog, AuditLogEvent, PromiseResult, UnconditionalUpdateOptions } from "."
 
 export default interface AuditLogDynamoGateway {
   create(message: AuditLog): PromiseResult<AuditLog>
   update(message: AuditLog): PromiseResult<AuditLog>
+  updateEntryUnconditionally(
```

Comment:
This function is very similar to `updateEntry` in the base class without having the version number.

I would assume it is to bypass the version number field. We use the version for [optimistic locking](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBMapper.OptimisticLocking.html) to prevent issues with concurrent writings to the same item.

Before we update the record we should get the current version from the audit log record by either getting the whole item or calling `getRecordVersion` in the base class.